compendium/LogServiceImpl/src/LogServiceImpl.cpp

```diff
@@ -0,0 +1,100 @@
+#include <sstream>
+
+#include "spdlog/sinks/stdout_color_sinks.h"
+#include "spdlog/spdlog.h"
+
+#include "cppmicroservices/ServiceReference.h"
+
+#include "LogServiceImpl.hpp"
+
+namespace cppmicroservices {
+namespace logservice {
+std::string GetExceptionMessage(const std::exception_ptr& ex)
+{
+  std::string message = "\nException logged: ";
+  if (ex) {
+    std::ostringstream stream;
+    try {
+      std::rethrow_exception(ex);
+    } catch (const std::exception& e) {
+      stream << typeid(e).name() << " : " << e.what();
+    }
+    message += stream.str();
+  } else {
+    message += "none";
+  }
+
+  return message;
+}
+
+std::string GetServiceReferenceInfo(const ServiceReferenceBase& sr)
+{
+  std::ostringstream stream;
+  stream << "\nServiceReference: " << sr;
```

Comment:
Same thing here...

std::string rval = “\nServiceReference: “ + sr;
return rval;

Reply:
We use the operator<< for the ServiceReferenceBase object to fill out the string. If we were to try and concatenate sr as you mentioned, this wouldn't compile.